### DISCUSSION TITLE
VFD setup

### DISCUSSION THREAD

**Discussion by mundsen** [terjeio/ioSender#321] [Q&A]
Hi
I have an Zoncn NZ100 VFD

Is there A file.. I can edit so I can control it from iosender/RS485?

I have found this page with some information about the parameters used by the VFD

https://blog.q1.se/2022/09/interfacing-with-the-zoncn-nz100-vfd-over-modbus/

https://www.jbcnc.se/get_file?product_id=952&file_id=397

Or is it better to swap my VFD with an Huanyang VFD ?

**Comment by terjeio** [comment]
First step is to try the MODVFD grblHAL [spindle driver](https://github.com/grblHAL/Plugins_spindle) and configure it according to the description in the first link above.
Settings $462 - $471 is used for that. With luck the [default values](https://github.com/grblHAL/Plugins_spindle/blob/88a9d51981ffe3f047f56b4eaf3ebf5e063bb6c3/vfd/spindle.c#L313-L324) should work.

Note that ioSender cannot control a VFD spindle directly, it always uses standard gcodes. These are translated to modbus commands by a spindle driver in the controller.

**Reply by mundsen** [reply]
Thank you I`ll try

I have different controllers (with and without RS485) - can I use an USB RS485 adapter if the controller do not have integrated RS485?

**Reply by andrewmarles** [reply]
You can't use a USB adapter connected to the host PC as it is the controller that runs the spindle, not IOSender.  What you can do if your controller has a serial UART exposed is use an external RS232 to RS485 converter.
